locations/jclouds/src/main/java/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java

```diff
@@ -225,41 +238,47 @@ void addPermissionsToLocation(Iterable<IpPermission> permissions, final String n
      *
+     * Notice that jclouds will attach 2 securityGroups to the node if the locationId is `aws-ec2` so it needs to
+     * look for the uniqueSecurityGroup rather than the shared securityGroup.
+     *
      * @param nodeId The id of the node in question
+     * @param locationId The id of the location in question
      * @param securityApi The API to use to list security groups
      * @return the security group unique to the given node, or null if one could not be determined.
      */
-    private SecurityGroup getUniqueSecurityGroupForNodeCachingSharedGroupIfPreviouslyUnknown(
-            String nodeId, SecurityGroupExtension securityApi) {
+    private SecurityGroup getUniqueSecurityGroupForNodeCachingSharedGroupIfPreviouslyUnknown(String nodeId, String locationId, SecurityGroupExtension securityApi) {
         Set<SecurityGroup> groupsOnNode = securityApi.listSecurityGroupsForNode(nodeId);
-        if (groupsOnNode.size() != 2) {
-            LOG.warn("Expected to find two security groups on node {} in app {} (one shared, one unique). Found {}: {}",
-                    new Object[]{nodeId, applicationId, groupsOnNode.size(), groupsOnNode});
-            return null;
-        }
-        String expectedSharedName = getNameForSharedSecurityGroup();
-        Iterator<SecurityGroup> it = groupsOnNode.iterator();
-        SecurityGroup shared = it.next();
         SecurityGroup unique;
-        if (shared.getName().endsWith(expectedSharedName)) {
-            unique = it.next();
-        } else {
-            unique = shared;
-            shared = it.next();
-        }
-        if (!shared.getName().endsWith(expectedSharedName)) {
-            LOG.warn("Couldn't determine which security group is shared between instances in app {}. Expected={}, found={}",
-                    new Object[]{applicationId, expectedSharedName, groupsOnNode});
-            return null;
-        }
-        // Shared entry might be missing if Brooklyn has rebound to an application
-        SecurityGroup old = sharedGroupCache.asMap().putIfAbsent(shared.getLocation(), shared);
-        LOG.info("Loaded unique security group for node {} (in {}): {}",
-                new Object[]{nodeId, applicationId, unique});
-        if (old == null) {
-            LOG.info("Proactively set shared group for app {} to: {}", applicationId, shared);
+        if (locationId.equals("aws-ec2")) {
+            if (groupsOnNode.size() != 2) {
+                LOG.warn("Expected to find two security groups on node {} in app {} (one shared, one unique). Found {}: {}",
+                        new Object[]{nodeId, applicationId, groupsOnNode.size(), groupsOnNode});
+                return null;
+            }
+            String expectedSharedName = getNameForSharedSecurityGroup();
+            Iterator<SecurityGroup> it = groupsOnNode.iterator();
+            SecurityGroup shared = it.next();
+            if (shared.getName().endsWith(expectedSharedName)) {
```

Comment:
This could be simplified a bit using `Iterables.tryFind(Iterable<T> iterable, Predicate<? super T> predicate)`